### ISSUE TITLE
/ & ? # are all doing the same thing if the parameter is a String

### ISSUE THREAD

**Issue by kszilagyi** [lemonlabsuk/scala-uri#8]
Hi,

So I have some code like this: `url & s"buildId=${n.i}"`
Where
`
url=https://teamcity.jetbrains.com/viewType.html?buildTypeId=ApacheAnt_BuildAntUsingMave
n.i=1255520
`
and 
the result is: `https://teamcity.jetbrains.com/viewType.html/buildId=1255520?buildTypeId=ApacheAnt_BuildAntUsingMave`

Now that's clearly not what I want. I think the problem is the combination of the implicit conversion of `String` to `Uri` and the fact that `&` is implemented like this:
```scala
def /(other: Uri) = merge(other)
def ?(other: Uri) = merge(other)
def `#`(other: Uri) = merge(other)
def &(other: Uri) = merge(other)
```
I guess the implicit conversion assumes it's a `Path` and then the merge will do the wrong thing. I guess an overload with `String` would solve the problem. What do you think?

**Comment by theon** [comment]
Thanks for raising; that is odd behaviour.

`?` and `&` currently do not support String. They were only ever designed to be used with a `Tuple`, like so:

```scala
import com.netaporter.uri.dsl._

val url = "https://teamcity.jetbrains.com/viewType.html?buildTypeId=ApacheAnt_BuildAntUsingMave"
val i = "1255520"

val myUrl = url & ("buildId" -> i)

myUrl: com.netaporter.uri.Uri = https://teamcity.jetbrains.com/viewType.html?buildTypeId=ApacheAnt_BuildAntUsingMave&buildId=1255520
```

We'd have to be careful adding support for Strings. Simply adding a overload for `?` that takes a String will break existing code, because `"http://example.com" ? "a" -> "b"` currently resolves to a Url, but adding a String overload, means it would resolve to a `Uri -> String`:

```scala
"http://example.com" ? "a" -> "b"
res0: (com.netaporter.uri.Uri, String) = (http://example.com?a,b)
```

For the meantime, I would recommend passing Tuples to `?` and `&`. I'll spend some time investigating what would be the best approach to stop the unexpected behaviour you experienced.

**Comment by theon** [comment]
What I said above turned out to be untrue. `"http://example.com" ? "a" -> "b"` has never been supported, so it was safe to add the functionality you wanted to use.

I have published a new version `0.5.1` and it should support your use case. Let me know if it works as expected.

**Comment by kszilagyi** [comment]
Thank you that was really quick. Did a quick look, seem to be working!
